File: code-coaches/medium/deja-vu/deja_vu.cpp
#include <iostream>

void check_deja_vu(std::string chars)
{
    char previousChar = ' ';
	bool isRepeated = false;

	for (auto &ch : chars)
	{
		if (ch == previousChar)
		{
			std::cout << "Deja Vu";
			isRepeated = true;
			break;
		}
		else
		{
			previousChar = ch;
		}
	}

	if (!isRepeated)
	{
		std::cout << "Unique";
	}
}

int main()
{
	std::string chars;
	std::cin >> chars;

    check_deja_vu(chars);

	return 0;
}

File: code-coaches/medium/deja-vu/deja_vu.py
def check_deja_vu(chars):
    previous_char = None
    is_repeated = False

    for char in chars:
        if char == previous_char:
            print("Deja Vu")
            is_repeated = True
            break
        else:
            previous_char = char

    if not is_repeated:
        print("Unique")


def main():
    input_str = input()
    check_deja_vu(input_str)


if __name__ == "__main__":
    main()
